bug: `tower-batch-control` compilation fails with `tokio_unstable`

### What happened?

Trying to compile `tokio-batch-control` with `--cfg=tokio_unstable`.

### What were you doing when the issue happened?

Trying to use `tokio-console` with Zallet.

### Zebra logs

```
error[E0308]: mismatched types
   --> /home/str4d/.cargo/git/checkouts/zebra-f970f3d516fe37ea/a77de50/tower-batch-control/src/service.rs:144:31
    |
144 |         batch.register_worker(worker_handle);
    |               --------------- ^^^^^^^^^^^^^ expected `JoinHandle<()>`, found `Result<JoinHandle<()>, Error>`
    |               |
    |               arguments to this method are incorrect
    |
    = note: expected struct `tokio::task::JoinHandle<_>`
                 found enum `Result<tokio::task::JoinHandle<_>, std::io::Error>`
note: method defined here
   --> /home/str4d/.cargo/git/checkouts/zebra-f970f3d516fe37ea/a77de50/tower-batch-control/src/service.rs:207:12
    |
207 |     pub fn register_worker(&mut self, worker_handle: JoinHandle<()>) {
    |            ^^^^^^^^^^^^^^^            -----------------------------
help: consider using `Result::expect` to unwrap the `Result<tokio::task::JoinHandle<()>, std::io::Error>` value, panicking if the value is a `Result::Err`
    |
144 |         batch.register_worker(worker_handle.expect("REASON"));
    |                                            +++++++++++++++++

    Checking abscissa_tokio v0.8.0
For more information about this error, try `rustc --explain E0308`.
error: could not compile `tower-batch-control` (lib) due to 1 previous error
```

### Zebra Version

_No response_

### Which operating systems does the issue happen on?

- [ ] Linux
- [ ] macOS
- [ ] Windows
- [ ] Other OS

### OS details

_No response_

### Additional information

The relevant code has been broken since `tokio 1.21.0` was released in September 2022, and was broken against the `tokio` repo five days before the code was initially merged (https://github.com/ZcashFoundation/zebra/pull/4750#discussion_r2104986670).

--- a/tower-batch-control/src/service.rs
+++ b/tower-batch-control/src/service.rs
@@ -137,6 +137,7 @@ where
             tokio::task::Builder::new()
                 .name(&format!("{} batch", batch_kind))
                 .spawn(worker.run().instrument(span))
+                .expect("panic on error to match tokio::spawn")
         };
         #[cfg(not(tokio_unstable))]
         let worker_handle = tokio::spawn(worker.run().instrument(span));

